Index: src/DeltaShell.Plugins.FMSuite.FlowFM/Model/WaterFlowFMModel.Running.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/DeltaShell.Plugins.FMSuite.FlowFM/Model/WaterFlowFMModel.Running.cs b/src/DeltaShell.Plugins.FMSuite.FlowFM/Model/WaterFlowFMModel.Running.cs
--- a/src/DeltaShell.Plugins.FMSuite.FlowFM/Model/WaterFlowFMModel.Running.cs	(revision 582812e96ea1581204d30711a95163de1f9c4248)
+++ b/src/DeltaShell.Plugins.FMSuite.FlowFM/Model/WaterFlowFMModel.Running.cs	(date 1609858605459)
@@ -57,7 +57,7 @@
         {
             runner.OnExecute();
         }
-
+        
         protected override void OnFinish()
         {
             runner.OnFinish();
@@ -73,6 +73,12 @@
             LogWarningWriteRestartModelRun();
         }
 
+        protected override void OnReset()
+        {
+            base.OnReset();
+            progressText = string.Empty;
+        }
+
         protected override void OnProgressChanged()
         {
             // Only update gui for every 1 percent progress (performance)
Index: src/DeltaShell.Plugins.DelftModels.HydroModel/HydroModel.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/DeltaShell.Plugins.DelftModels.HydroModel/HydroModel.cs b/src/DeltaShell.Plugins.DelftModels.HydroModel/HydroModel.cs
--- a/src/DeltaShell.Plugins.DelftModels.HydroModel/HydroModel.cs	(revision 582812e96ea1581204d30711a95163de1f9c4248)
+++ b/src/DeltaShell.Plugins.DelftModels.HydroModel/HydroModel.cs	(date 1609855480422)
@@ -245,6 +245,12 @@
         {
             get
             {
+                // Force reset when model is being reinitialized
+                if (Status == ActivityStatus.Initializing || Status == ActivityStatus.None)
+                {
+                    return GetProgressTextCore(0);
+                }
+                
                 TimeDependentModelBase[] timeDependentModelBases = Activities.GetActivitiesOfType<TimeDependentModelBase>().ToArray();
                 return timeDependentModelBases.Any()
                            ? GetProgressTextCore(timeDependentModelBases.Average(m => m.ProgressPercentage))
